extensions/wikia/MercuryApi/models/MercuryApi.class.php

```diff
@@ -263,27 +263,37 @@ public function processCuratedContent( $data ) {
 		}
 
 		if ( $process ) {
-			foreach ( $data[ $process ] as &$item ) {
-				$item = $this->addArticleUrlToCuratedContentItem( $item );
+			$items = [];
+			foreach ( $data[ $process ] as $item ) {
+				$processedItem = $this->processCuratedContentItem( $item );
+				if ( !empty( $processedItem ) ) {
+					$items[] = $processedItem;
+				}
 			}
+			$data[ $process ] = $items;
 		}
 
 		return $data;
 	}
 
 	/**
-	 * @desc Mercury can't open article using ID - we need to create a local link
+	 * @desc Mercury can't open article using ID - we need to create a local link.
+	 * If article doesn't exist (Title is null) return null.
+	 * In other case return item with updated article_local_url.
 	 * TODO Implement cache for release version.
 	 * Platform Team is OK with hitting DB for MVP (10-15 wikis)
 	 *
 	 * @param $item
 	 * @return mixed
 	 */
-	private function addArticleUrlToCuratedContentItem( $item ) {
+	private function processCuratedContentItem( $item ) {
 		if ( !empty( $item[ 'article_id' ] ) ) {
-			$item[ 'article_local_url' ] = Title::newFromID( $item[ 'article_id' ] )->getLocalURL();
+			$title = Title::newFromID( $item[ 'article_id' ] );
```

Comment:
there is a comment above to check with platform team when we would go 100% or even for phase 2.